compiler-plugin/src/main/scala/scalapb/compiler/GrpcServicePrinter.scala

```diff
@@ -179,12 +180,18 @@ final class GrpcServicePrinter(service: ServiceDescriptor, implicits: Descriptor
 
     val grpcMethodDescriptor = "_root_.io.grpc.MethodDescriptor"
 
+    val idempotencyLevel = method.getOptions.getIdempotencyLevel
+    val safe             = idempotencyLevel == IdempotencyLevel.NO_SIDE_EFFECTS
+    val idempotent       = idempotencyLevel == IdempotencyLevel.IDEMPOTENT
+
     p.add(
       s"""${method.deprecatedAnnotation}val ${method.grpcDescriptor.nameSymbol}: $grpcMethodDescriptor[${method.inputType.scalaType}, ${method.outputType.scalaType}] =
          |  $grpcMethodDescriptor.newBuilder()
          |    .setType($grpcMethodDescriptor.MethodType.$methodType)
          |    .setFullMethodName($grpcMethodDescriptor.generateFullMethodName("${service.getFullName}", "${method.getName}"))
          |    .setSampledToLocalTracing(true)
+         |    .setSafe($safe)
```

Comment:
Can we `setSafe` and `setIdempotent` only when `method.getOptions.hasIdempotencyLevel` ? You cal use `p.when` to emit code conditionally.

Reply:
Thanks for spotting it. Also found the tests and added a test for this case